ethereumj-core/src/main/java/org/ethereum/sharding/processing/state/ActiveState.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) [2016] [ <ether.camp> ]
+ * This file is part of the ethereumJ library.
+ *
+ * The ethereumJ library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ethereumJ library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ethereumJ library. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.sharding.processing.state;
+
+import org.ethereum.datasource.Serializer;
+import org.ethereum.util.ByteUtil;
+import org.ethereum.util.RLP;
+import org.ethereum.util.RLPElement;
+import org.ethereum.util.RLPList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.ethereum.crypto.HashUtil.blake2b;
+import static org.ethereum.sharding.processing.consensus.BeaconConstants.CYCLE_LENGTH;
+import static org.ethereum.util.ByteUtil.isSingleZero;
+import static org.ethereum.util.ByteUtil.toHexString;
+
+/**
+ * Active beacon chain state
+ */
+public class ActiveState {
```

Comment:
If we introduce a new state it may be, also, reasonable to add a transition of that state

Reply:
Also we don't have separate transition for `CrystallizedState` and it's done in `BeaconStateTransition`. I'd add `ActiveState` changes there too when it is needed as it's in conjunction with `CrystallizedState` changes and both are parts of `BeaconStateTransition`